feat(rocsparse): add rocsparse_diagonal_mode plumbing (#10981)
ROCm/rocm-libraries: projects/rocsparse/CHANGELOG.md

* Added the `rocsparse_dnvec_descr_create_scalar` auxiliary routine, which creates a size-one dense vector descriptor for a host or device scalar.
* Added batched support to the SpMM algorithm `rocsparse_spmm_alg_csr_nnz_split` and `rocsparse_spmm_alg_csr_merge_path`.
* Added `rocsparse_sddmm` batched support to CSR, CSC, COO, COO AoS, and ELL formats.

> * Added the `rocsparse_diagonal_mode` enum and the `rocsparse_sptrsv_input_diagonal_mode` / `rocsparse_sptrsm_input_diagonal_mode` set-input values, which make `rocsparse_sptrsv` and `rocsparse_sptrsm` run a diagonal-only solve. The CSR and CSC formats are supported.

Reviewer: ```suggestion
* Added the `rocsparse_diagonal_mode` enum and the `rocsparse_sptrsv_input_diagonal_mode` and `rocsparse_sptrsm_input_diagonal_mode` set-input values to enable diagonal-only solves in `rocsparse_sptrsv` and `rocsparse_sptrsm`. CSR and CSC formats are supported.
```
